src/clients/mnemonic/client.ts

```diff
@@ -0,0 +1,154 @@
+import type _algosdk from "algosdk";
+import Algod, { getAlgodClient } from "../../algod";
+import BaseWallet from "../base";
+import { DEFAULT_NETWORK, PROVIDER_ID } from "../../constants";
+import type { Wallet, TransactionsArray, Network } from "../../types";
+import { ICON } from "./constants";
+import { InitParams, MnemonicWalletClientConstructor } from "./types";
+import algosdk from "algosdk";
+
+class MnemonicWalletClient extends BaseWallet {
+  #client?: _algosdk.Account;
+  id: PROVIDER_ID;
+  network: Network;
+
+  constructor({
+    metadata,
+    id,
+    algosdk,
+    algodClient,
+    network,
+  }: MnemonicWalletClientConstructor) {
+    super(metadata, algosdk, algodClient);
+
+    this.id = id;
+    this.network = network;
+    this.metadata = MnemonicWalletClient.metadata;
+  }
+
+  static metadata = {
+    id: PROVIDER_ID.MNEMONIC,
+    name: "MNEMONIC",
+    icon: ICON,
+    isWalletConnect: false,
+  };
+
+  static async init({
+    algodOptions,
+    algosdkStatic,
+    network = DEFAULT_NETWORK,
+  }: InitParams) {
+    try {
+      const algosdk = algosdkStatic || (await Algod.init(algodOptions)).algosdk;
+      const algodClient = await getAlgodClient(algosdk, algodOptions);
+      console.log(network, algodClient);
+      return new MnemonicWalletClient({
+        metadata: MnemonicWalletClient.metadata,
+        id: PROVIDER_ID.MNEMONIC,
+        algosdk: algosdk,
+        algodClient: algodClient,
+        network,
+      });
+    } catch (e) {
+      console.error("Error initializing...", e);
+      return null;
+    }
+  }
+
+  async connect(): Promise<Wallet> {
+    const password = await this.requestPassword();
+
+    if (!password) {
+      this.#client = undefined;
+      throw new Error("Mnemonic passphrase is required");
+    }
+
+    this.#client = algosdk.mnemonicToSecretKey(password);
+
+    return {
+      ...MnemonicWalletClient.metadata,
+      accounts: [
+        {
+          name: `MnemonicWallet 1`,
+          address: this.#client.addr,
+          providerId: MnemonicWalletClient.metadata.id,
+        },
+      ],
+    };
+  }
+
+  async disconnect() {
+    this.#client = undefined;
+    return;
+  }
+
+  async reconnect(): Promise<Wallet | null> {
+    return null;
+  }
+
+  async requestPassword(): Promise<string> {
+    // TODO: store it locally?
+    const pass = prompt("enter mnemonic passphrase, 25 words");
+    return pass ? pass : "";
+  }
+
+  signTransactions(
+    connectedAccounts: string[],
+    transactions: Uint8Array[]
```

Comment:
```suggestion
    connectedAccounts: string[],
    transactions: Uint8Array[],
    indexesToSign?: number[],
    returnGroup = true
```

Add the `indexesToSign` and `returnGroup` param. 

The `indexesToSign` param lets you specify an array of indexes in the atomic group that should be signed. It's modeled after the `Signer` interface specified here: https://github.com/algorand/js-algorand-sdk/blob/develop/src/signer.ts#L15-L18

The `returnGroup` option sets whether or not the entire transaction group is returned, or only the transactions that were signed. Defaults to `true` to return the entire group.